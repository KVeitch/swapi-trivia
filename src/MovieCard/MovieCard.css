.movie__card {
    background-color: rgba(69, 141, 150, 0.77);
    border-radius: 5px;
    margin: 1.5%;
    opacity: 1;
    flex-direction: column;
    display: flex;
    min-width: 26vh;
    padding: 4px;
    justify-content: space-evenly;
    height:48vh;
}

.card__title {
    height:100px;
    font-family: 'SpaceMeatballRegular';
    font-weight: normal;
    font-style: normal;
    text-shadow: 3px 4px 4px rgba(23, 6, 1, 0.94)
}

.card__episode {
    height:60px;
}

.card__release {
    height:60px;
}

.characters__button {
    background-color: rgba(62, 110, 116, 0.99);
    border: none;
    border-radius: 5px;
    box-shadow: 0 10px 6px -6px rgba(41, 72, 76, 0.99);
    color: white;
    font-size: 1em;
    width: 100%;
    height:60px;
    margin-bottom: 20px;
}