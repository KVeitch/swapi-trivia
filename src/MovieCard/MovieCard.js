import React, { Component } from 'react';
import './MovieCard.css';
import CharacterCard from '../CharacterCard/CharacterCard';
import { createCharacterList } from '../apiCalls';

class MovieCard extends Component {
  constructor() {
    super();
    this.state = {
      showCharacterCard : false,
      characters : []
    }
  }

  componentDidMount() {
    createCharacterList(this.props.characters)
    .then(data=>this.setState({characters:data}))
  }

  viewCharacter = (e) => {
    this.setState({ showCharacterCard : !this.state.showCharacterCard})
    this.props.changeSelectedMovie(e.target.value)
  }

  render() {

    const characters = this.state.showCharacterCard ? 
    this.state.characters.map((character, i) => {
      return <CharacterCard key={i} character={character}/>
    })
    : '';
  

    return (
      <div className='movie-card'>
        <h2>{this.props.title}</h2>
        <p>Episode {this.props.episode}</p>
        <p>Released in {this.props.releaseYear.slice(0, 4)}</p>
        <button onClick={this.viewCharacter} className='characters-button' value={this.props.episode}>View Characters</button>
        {characters}
      </div>
    ) 
  }
}
export default MovieCard;