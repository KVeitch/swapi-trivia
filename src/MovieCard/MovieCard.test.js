import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from './MovieCard';

describe('Movie Card', () => {
    let wrapper;
    let mockChangeSelectedMovie = jest.fn()
    it('should match the snapshot with all the data passed in correctly', () => {
        wrapper =
            shallow(<MovieCard
                title={''}
                episode={3}
                img={''}
                key={3}
                releaseYear={''}
                characters={[]}
                changeSelectedMovie={mockChangeSelectedMovie}
                id={3}/>)
        expect(wrapper).toMatchSnapshot();
    });

    it.skip('should fire changeSelectedMovie when the View Characters Button is clicked', () => {
        wrapper.find('.characters__button').simulate('onClick');
        wrapper.instance().changeSelectedMovie();
        expect(wrapper.instance().changeSelectedMovie).toHaveBeenCalled();
    });
});