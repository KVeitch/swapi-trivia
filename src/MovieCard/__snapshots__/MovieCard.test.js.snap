// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="movie__card"
>
  <h2
    className="card__title"
  />
  <p
    className="card__episode"
  >
    Episode 
    3
  </p>
  <img
    alt="movie poster for "
    className="movie_image"
  />
  <p
    className="card__release"
  >
    Released in 
  </p>
  <Link
    to="/movies/3"
  >
    <button
      className="characters__button"
      id={3}
      onClick={[Function]}
    >
      View Characters
    </button>
  </Link>
</div>
`;
